pkg/pulp_client/task.go

```diff
@@ -0,0 +1,115 @@
+package pulp_client
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"time"
+
+	zest "github.com/content-services/zest/release/v3"
+	"github.com/rs/zerolog/log"
+	"golang.org/x/exp/slices"
+)
+
+const (
+	COMPLETED string = "completed"
+	WAITING   string = "waiting"
+	RUNNING   string = "running"
+	SKIPPED   string = "skipped"
+	CANCELED  string = "canceled"
+	FAILED    string = "failed"
+)
+
+// GetTask Fetch a pulp task
+func (r pulpDaoImpl) GetTask(taskHref string) (zest.TaskResponse, error) {
+	task, _, err := r.client.TasksApi.TasksRead(r.ctx, taskHref).Execute()
+
+	if err != nil {
+		return zest.TaskResponse{}, err
+	}
+
+	return *task, nil
+}
+
+// PollTask Poll a task and return the final task object
+func (r pulpDaoImpl) PollTask(taskHref string) (*zest.TaskResponse, error) {
+	var task zest.TaskResponse
+	var err error
+	syncInProgress := true
```

Comment:
Is this still representative of just "syncInProgress" or did it evolve into "taskInProgress"? I see publication and distribution also call `PollTask`, so I'm guessing the meaning has changed. If that's true I would rename it.